src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -55,12 +55,18 @@ type Client struct {
 	rp eventual.Value
 
 	l net.Listener
+
+	proxyAll func() bool
```

Comment:
Just store the proxyAll function in the struct instead of re-configuring the value every time the config changes. Eliminates threading issues as well as simplifies the code.